cli/cmd/install.go

```diff
@@ -33,156 +30,99 @@ This command provides the Kubernetes configs necessary to install the Buoyant
 Cloud Agent.
 
 If an agent is not already present on the current cluster, this command
-redirects the user to Buoyant Cloud to set up a new agent. Once the new agent is
-set up, this command will output the agent manifest.
+will provide you with a manifest that should perform auto registration of
+the agent with the name you have specified.
 
 If an agent is already present, this command retrieves an updated manifest from
-Buoyant Cloud and outputs it.`,
-		Example: `  # Default install.
+Buoyant Cloud and outputs it.
+
+Note that this command required CLIENT_ID and CLIENT_SECRET env vars to be set.
+To get a CLIENT_ID and CLIENT_SECRET, log in to https://buoyant.cloud.`,
+		Example: `  # Default install (no agent on cluster).
+  linkerd buoyant install --agent-name=my-new-agent | kubectl apply -f -
+
+  # Obtain a manifest for an agent that already exists on your cluster
   linkerd buoyant install | kubectl apply -f -
 
   # Install onto a specific cluster
   linkerd buoyant --context test-cluster install | kubectl --context test-cluster apply -f -`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			client, err := k8s.New(cfg.kubeconfig, cfg.kubecontext, cfg.bcloudServer)
+			client, err := k8s.New(installCfg.kubeconfig, cfg.kubecontext)
 			if err != nil {
 				return err
 			}
 
-			return install(cmd.Context(), cfg, client, browser.OpenURL)
+			// get clientID and clientSecret from either k8s or env vars
+			var clientID, clientSecret string
+
+			secret, err := client.Secret(cmd.Context())
+			if err == nil && secret != nil {
+				clientID = string(secret.Data["client_id"])
+				clientSecret = string(secret.Data["client_secret"])
+			}
+
+			// prefer env vars
+			if os.Getenv("CLIENT_ID") != "" {
+				clientID = os.Getenv("CLIENT_ID")
+			}
+			if os.Getenv("CLIENT_SECRET") != "" {
+				clientSecret = os.Getenv("CLIENT_SECRET")
+			}
+
+			if clientID == "" {
+				return errors.New("install command requires setting CLIENT_ID")
+			}
+			if clientSecret == "" {
+				return errors.New("install command requires setting CLIENT_SECRET")
+			}
+
+			apiClient := bcloudapi.New(clientID, clientSecret, installCfg.bcloudAPI, !installCfg.noTLS)
+
+			return install(cmd.Context(), installCfg, client, apiClient)
 		},
 	}
 
+	cmd.Flags().StringVar(&installCfg.agentName, "agent-name", "", "The name of the agent.")
```

Comment:
In the bcloud UI, we refer to these strings as clusters, not agents, so it might be worth renaming this flag to `--cluster-name` (let's discuss at standup).

Reply:
We can do that, but then do we follow this convention in the code as well? To me cluster-name is a bit confusing as it might be mistaken for the name of the context... We can go either way